alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportGetAlertsAction.kt

```diff
@@ -115,50 +116,30 @@ class TransportGetAlertsAction @Inject constructor(
                     .from(tableProp.startIndex)
 
             client.threadPool().threadContext.stashContext().use {
-                resolve(getAlertsRequest, searchSourceBuilder, actionListener)
+                resolve(searchSourceBuilder, actionListener)
             }
         }
 
         fun resolve(
-            getAlertsRequest: GetAlertsRequest,
             searchSourceBuilder: SearchSourceBuilder,
             actionListener: ActionListener<GetAlertsResponse>
         ) {
-            // auth header is null when: 1/ security is disabled. 2/when user is super-admin.
-            if (getAlertsRequest.authHeader.isNullOrEmpty()) {
-                // auth header is null when: 1/ security is disabled. 2/when user is super-admin.
+            // user is null when: 1/ security is disabled. 2/when user is super-admin.
+            if (user == null) {
+                // user is null when: 1/ security is disabled. 2/when user is super-admin.
                 search(searchSourceBuilder, actionListener)
             } else if (!filterByEnabled) {
                 // security is enabled and filterby is disabled.
                 search(searchSourceBuilder, actionListener)
             } else {
                 // security is enabled and filterby is enabled.
-                val authRequest = AuthUserRequestBuilder(
-                        getAlertsRequest.authHeader
-                ).build()
-                restClient.performRequestAsync(authRequest, object : ResponseListener {
-                    override fun onSuccess(response: Response) {
-                        try {
-                            val user = User(response)
-                            addFilter(user, searchSourceBuilder, "monitor_user.backend_roles")
-                            log.info("Filtering result by: ${user.backendRoles}")
-                            search(searchSourceBuilder, actionListener)
-                        } catch (ex: IOException) {
-                            actionListener.onFailure(AlertingException.wrap(ex))
-                        }
-                    }
-
-                    override fun onFailure(ex: Exception) {
-                        when (ex.message?.contains("Connection refused")) {
-                            // Connection is refused when security plugin is not present. This case can happen only with integration tests.
-                            true -> {
-                                addFilter(User(), searchSourceBuilder, "monitor_user.backend_roles")
-                                search(searchSourceBuilder, actionListener)
-                            }
-                            false -> actionListener.onFailure(AlertingException.wrap(ex))
-                        }
-                    }
-                })
+                try {
+                    addFilter(user as User, searchSourceBuilder, "monitor_user.backend_roles")
```

Comment:
Is alerting using `monitor_user` or `user` ?

Reply:
Should we use `user.backend_roles.keyword` to match the role exactly?

Reply:
This is alert. Alert has information on related monitor. `monitor_user` is related monitor user. If you see monitor its `monitor.user.backend_roles`.

Reply:
Cool. got it. Any comments for the second question ?
> Should we use user.backend_roles.keyword to match the role exactly?

Reply:
Fixed in this PR https://github.com/opendistro-for-elasticsearch/alerting/pull/290